genie-web/src/main/java/com/netflix/genie/web/properties/LocalAgentLauncherProperties.java

```diff
@@ -45,11 +45,42 @@
      */
     public static final String PROPERTY_PREFIX = "genie.agent.launcher.local";
 
+    /**
+     * Placeholder for server port in command-line-template.
+     */
+    public static final String SERVER_PORT_PLACEHOLDER = "<SERVER_PORT_PLACEHOLDER>";
+
+    /**
+     * Placeholder for job id in command-line-template.
+     */
+    public static final String JOB_ID_PLACEHOLDER = "<JOB_ID_PLACEHOLDER>";
+
+    /**
+     * Placeholder for agent jar path in command-line-template.
+     */
+    public static final String AGENT_JAR_PLACEHOLDER = "<AGENT_JAR_PLACEHOLDER>";
+
     /**
      * The command that should be run to execute the Genie agent. Required.
      */
-    @NotEmpty(message = "The agent executable cannot be empty or it will be impossible to launch an agent process")
-    private List<@NotBlank String> executable = Lists.newArrayList("java", "-jar", "/tmp/genie-agent.jar");
+    @SuppressWarnings("PMD.AvoidUsingHardCodedIP")
+    @NotEmpty(message = "The agent command-line template cannot be empty")
+    private List<@NotBlank String> agentLaunchCommandTemplate = Lists.newArrayList(
+        "java",
+        "-jar", AGENT_JAR_PLACEHOLDER,
+        "exec",
+        "--serverHost", "127.0.0.1",
+        "--serverPort", SERVER_PORT_PLACEHOLDER,
+        "--api-job",
+        "--no-cleanup",
```

Comment:
I doubt we want `--no-cleanup` as the default going forward we might want to take it out now. I had it in for testing. Which we could do by overriding this in a `integration` test profile